From 0b247aa2f8ef5653e80342a994a72f6dccdb0ae4 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Wed, 15 Dec 2021 18:49:23 +0100
Subject: [PATCH 019/954] feat(stm32mp1-fdts): add MCE support for STM32MP13 DK
 board

Add an st-mem-encrypt node in a dedicated file. This file will be
included by boards fw-config file if they support encryption.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git b03c7a25f20cfd3a491075ee974c925a4ff00427 feat(stm32mp1-fdts): add MCE support for STM32MP13 DK board]
Change-Id: I7d2b278da144a10c10d194d0475ab11b1bc142bb
Signed-off-by: Yann Gautier <yann.gautier@st.com>
---
 fdts/stm32mp13-fw-config-mem-encrypt.dtsi | 13 +++++++++++++
 fdts/stm32mp135f-dk-fw-config.dts         |  1 +
 2 files changed, 14 insertions(+)
 create mode 100644 fdts/stm32mp13-fw-config-mem-encrypt.dtsi

diff --git a/fdts/stm32mp13-fw-config-mem-encrypt.dtsi b/fdts/stm32mp13-fw-config-mem-encrypt.dtsi
new file mode 100644
index 0000000000..89e2d2431f
--- /dev/null
+++ b/fdts/stm32mp13-fw-config-mem-encrypt.dtsi
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (c) 2021, STMicroelectronics - All Rights Reserved
+ */
+
+#include <dt-bindings/soc/stm32mp13-mce.h>
+
+/ {
+	st-mem-encrypt {
+		compatible = "st,mem-encrypt";
+		memory-ranges = <DDR_SEC_BASE DDR_SEC_SIZE MCE_ENCRYPT>;
+	};
+};
diff --git a/fdts/stm32mp135f-dk-fw-config.dts b/fdts/stm32mp135f-dk-fw-config.dts
index 21f82422b3..41ebf73c21 100644
--- a/fdts/stm32mp135f-dk-fw-config.dts
+++ b/fdts/stm32mp135f-dk-fw-config.dts
@@ -5,3 +5,4 @@
 
 #define DDR_SIZE	0x20000000 /* 512MB */
 #include "stm32mp13-fw-config.dtsi"
+#include "stm32mp13-fw-config-mem-encrypt.dtsi"
-- 
2.53.0

